resources/assets/coffee/react/mp-history/main.coffee

```diff
@@ -64,7 +74,8 @@ class MPHistory.Main extends React.Component
         events: newEvents
         allEventsCount: data.all_events_count
         users: newUsers
-        since: _.last(newEvents).id
+        since: if !lastGameEvent? || lastGameEvent.game.data.end_time then lastEvent.id else lastGameEvent.id - 1
+        lastGameId: if lastGameEvent? then lastGameEvent.id else 0
         disbanded: lastEvent.event_type == 'match-disbanded'
 
     .always =>
```

Comment:
and the timeout should be stored in a variable so it can be cleared during unmount to avoid running it after unmounted.